src/main/java/com/xeeva/catalog/CostCenterPage.java

```diff
@@ -1,438 +1,448 @@
+			TestReporter.logStep("itemNumber :"+ itemNumber);
+			ddCCLineLevel.get(0).select(CC);
+			System.out.println(verifyCostCenter(changeType,itemNumber,CC,null));
+			TestReporter.assertTrue(verifyCostCenter(changeType,itemNumber,CC,null),"Cost Center Updated at Line Level Successfully!!");
+			break;
+
+		case "headerlevel" :
+			lstCostCenterHeaderLevel.select(CC);
+			//Handle Alert if present
+			if(AlertHandler.isAlertPresent(driver, 6)){
+				AlertHandler.handleAlert(driver, 6);
+			}
+			System.out.println(verifyCostCenter(changeType,null,CC,null));
+			TestReporter.assertTrue( verifyCostCenter(changeType,null,CC,null),"Cost Center Updated at Header Level Successfully!!");
+			break;
+
+		default : System.out.println();
+		}
+	}
+
+	public String selectCCToCopyItem(){
+		String ccValue = lstSelectCCValue.get(0).getText();
+		System.out.println("CostCenter Name "+lstSelectCCValue.get(0).getText());
+		driver.executeJavaScript("arguments[0].click();", lstSelectCC.get(0));
+		Sleeper.sleep(5000);
+		return ccValue;
+	}
+
+	public String readItemNuber(){
+		return lstItemNoGrid.get(0).getText();
+	}
+
+
+
+	/**
+	 * @summary: Method to get the RequiredByDate from textbox at CC_line-level.
+	 * @author: Praveen Namburi, @Version: Created 21-09-2016
+	 * @return getReqByDate
+	 */
+	public String getReqByDateAtLineLevel(){
+		String getRequreDate = "";
+		List<WebElement> reqByDates = dateCCInputLineLevel;
+		for(WebElement reqByDate : reqByDates){
+			String getReqByDate_beforeChange1 = reqByDate.getAttribute("value");
+			//System.out.println(getReqByDate_beforeChange1);
+			getRequreDate=getReqByDate_beforeChange1;
+			break;
+		}
+		return getRequreDate;
+	}
+
+	/**
+	 * @summary: Method to verify the RequiredBydate has been updated Successfully at CC_level.
+	 * @author: Praveen Namburi, @version: Created 21-09-2016.
+	 */
+	public void verify_RequiredByDateUpdated_atCCLevel(){
+
+		//Added wait statement to wait until the Cart item added successfull message to be displayed.
+		WebDriverWait wait = new WebDriverWait(driver,10);
+		WebElement lblCartAddItemMessage =wait.until(ExpectedConditions.
+				visibilityOfElementLocated(By.xpath("//div[@id='divAppInfoMsg'][@class='addMessage']")));
+		String getReqByDateMessage = lblCartItemAddedMessage.getText();
+		TestReporter.logStep("Message after changing the RequiredBy date at CC_level : "+ getReqByDateMessage);
+		TestReporter.assertTrue(getReqByDateMessage.contains("date has been updated."), 
+				"Required by date has been updated at CC-Level.");
+
+	}
+
+	/**
+	 * @summary: Method to change the RequiredBy value at linelevel.
+	 * @author: Praveen Namburi, @version: Created 20-09-2016.
+	 */
+	public void change_RequiredByAtLineLevel(){
+		// after cart check out, application taking time to load cost center page 
+		lblCC.syncVisible(30, false);
+		lblCC.isDisplayed();
+		driver.setElementTimeout(Constants.ELEMENT_TIMEOUT);
+
```

Comment:
Need to remove line 358, if it is not returning any value and doing something with it.   ALso need to remove the setElementTimeout, as you are setting it to the default - which it should already be at.